presto-main/src/main/java/io/prestosql/split/SplitManager.java

```diff
@@ -36,10 +40,15 @@
     private final ConcurrentMap<ConnectorId, ConnectorSplitManager> splitManagers = new ConcurrentHashMap<>();
     private final int minScheduleSplitBatchSize;
 
+    // This is used to fetch a table layout if the TableHandle doesn't have one set
+    // It's a temporary measure until we get rid of TableLayouts from the SPI
+    private final Metadata metadata;
```

Comment:
How it can have unset `TableLayout`? Doesn't that indicate error is planning?

Reply:
LocalQueryRunner doesn't run AddExchanges (https://github.com/prestosql/presto/issues/275), so if there's no filter that can be pushed down there won't be a layout associated with the TableHandle in the TableScan by the time it gets here.

Reply:
Hm. Could we change `LocalQueryRunner` instead of `SplitManager`? `SplitManager` seems more essential than `LocalQueryRunner` to the engine so it should have more solid design.

Maybe we should choose some layout anyway by the end of planning? E.g:
in `PlanOptimizers` you could have some rule for that when `forceSingleNode==true`. Then we could also have a validator that layout is always present.

Reply:
That's a much bigger (non-essential) change, but let me see if I can come up with a way to avoid needing Metadata here.

Reply:
Found a much simpler way to do it that keeps the hack inside LocalQueryRunner (which is already hacky due to the issue I linked above). Take a look: https://github.com/prestosql/presto/pull/363/commits/500156a7515bc73b50f932ba44e04dc804ed8eee

Thanks for prompting me to think harder about a better way to do it :)